	lazy_matrix_mul module Test
======================================
# This is a doctest for the module lazy_matrix_mul

#import the module
>>> lazy_matrix_mul = __import__('101-lazy_matrix_mul').lazy_matrix_mul

#Test Cases

>>> lazy_matrix_mul([[1, 2], [3, 4]], [[1, 2], [3, 4]])
array([[ 7, 10],
       [15, 22]])
>>> lazy_matrix_mul([[1, 2]], [[3, 4], [5, 6]])
array([[13, 16]])

>>> lazy_matrix_mul([[1, 2], [-3, -4]], [[-5, -6], [7, 8]])
array([[  9,  10],
       [-13, -14]])
>>> lazy_matrix_mul(2, [[-5, -6], [7, 8]])
Traceback (most recent call last):
...
ValueError: Scalar operands are not allowed, use '*' instead

>>> lazy_matrix_mul([[-5, -6], [7, 8]], 6)
Traceback (most recent call last):
...
ValueError: Scalar operands are not allowed, use '*' instead

>>> lazy_matrix_mul([[4,5], [3, 9], 8], [[-5, -6], [7, 8]])
Traceback (most recent call last):
...
ValueError: setting an array element with a sequence.

>>> lazy_matrix_mul([[-5, -6], [7, 8]], [[4,5], [3, 9], 8])
Traceback (most recent call last):
...
ValueError: setting an array element with a sequence.

>>> lazy_matrix_mul([], [[-5, -6], [7, 8]])
Traceback (most recent call last):
...
ValueError: shapes (0,) and (2,2) not aligned: 0 (dim 0) != 2 (dim 0)

>>> lazy_matrix_mul([[]], [[-5, -6], [7, 8]])
Traceback (most recent call last):
...
ValueError: shapes (1,0) and (2,2) not aligned: 0 (dim 1) != 2 (dim 0)

>>> lazy_matrix_mul([[-5, -6], [7, 8]], [])
Traceback (most recent call last):
...
ValueError: shapes (2,2) and (0,) not aligned: 2 (dim 1) != 0 (dim 0)

>>> lazy_matrix_mul([[-5, -6], [7, 8]], [[]])
Traceback (most recent call last):
...
ValueError: shapes (2,2) and (1,0) not aligned: 2 (dim 1) != 1 (dim 0)

>>> lazy_matrix_mul([[3, '5', 4], [4, 7, 8]], [[-5, -6], [7, 8]])
Traceback (most recent call last):
...
ValueError: shapes (2,3) and (2,2) not aligned: 3 (dim 1) != 2 (dim 0)

>>> lazy_matrix_mul([[-5, -6], [7, 8]], [[3, '5', 4], [4, 7, 8]])
Traceback (most recent call last):
...
TypeError: invalid data type for einsum

>>> lazy_matrix_mul([[3, 5, 4], [4, 7, 8]], [[-5, -6], [7, 8]])
Traceback (most recent call last):
...
ValueError: shapes (2,3) and (2,2) not aligned: 3 (dim 1) != 2 (dim 0)

>>> lazy_matrix_mul()
Traceback (most recent call last):
...
TypeError: lazy_matrix_mul() missing 2 required positional arguments: 'm_a' and 'm_b'


>>> lazy_matrix_mul([[3,4]])
Traceback (most recent call last):
...
TypeError: lazy_matrix_mul() missing 1 required positional argument: 'm_b'


>>> lazy_matrix_mul([[3,4], [4,2,5]], [[2, 5,], [8, 4]])
Traceback (most recent call last):
...
ValueError: setting an array element with a sequence.


>>> lazy_matrix_mul([[3,4], [4,5]], [[2, 5, 8], [8, 4]])
Traceback (most recent call last):
...
ValueError: setting an array element with a sequence.
